keyboard.c

```diff
@@ -224,42 +223,54 @@ ps2_scancode_from_SDL_Scancode(SDL_Scancode scancode)
 void
 handle_keyboard(bool down, SDL_Keycode sym, SDL_Scancode scancode)
 {
-	if (down) {
-		if (log_keyboard) {
-			printf("DOWN 0x%02X\n", scancode);
-			fflush(stdout);
-		}
+    if (down) {
+#if __APPLE__
+#if TARGET_IPHONE_SIMULATOR || TARGET_OS_IPHONE
+        
+        // iOS device
+        if (scancode == SDL_SCANCODE_MENU) {
+            sendNotification("ShowKeyboard");
+        }
+#elif TARGET_OS_MAC
+        // Other kinds of Mac OS
+#else
+#   error "Unknown Apple platform"
+#endif
+#endif
+        
+        if (log_keyboard) {
+                printf("DOWN 0x%02X\n", scancode);
+                fflush(stdout);
+            }
 
-		int ps2_scancode = ps2_scancode_from_SDL_Scancode(scancode);
-		if (ps2_scancode == 0xff) {
-			// "Pause/Break" sequence
-			ps2_buffer_add(0, 0xe1);
-			ps2_buffer_add(0, 0x14);
-			ps2_buffer_add(0, 0x77);
-			ps2_buffer_add(0, 0xe1);
-			ps2_buffer_add(0, 0xf0);
-			ps2_buffer_add(0, 0x14);
-			ps2_buffer_add(0, 0xf0);
-			ps2_buffer_add(0, 0x77);
-		} else {
-			if (ps2_scancode & EXTENDED_FLAG) {
-				ps2_buffer_add(0, 0xe0);
-			}
-			ps2_buffer_add(0, ps2_scancode & 0xff);
-		}
-	} else {
-		if (log_keyboard) {
-			printf("UP   0x%02X\n", scancode);
-			fflush(stdout);
-		}
+            int ps2_scancode = ps2_scancode_from_SDL_Scancode(scancode);
```

Comment:
The indentation changed here. In Xcode, you can change it to tabs by going to Preferences -> Text Editing -> Indentation. I think you can also do it by project somewhere.

Reply:
👍